5.5/run-mysqld.sh

```diff
@@ -94,6 +103,29 @@ if [ "$1" = "mysqld" -a ! -d "$MYSQL_DATADIR/mysql" ]; then
 		EOSQL
 	fi
 	mysqladmin $admin_flags flush-privileges shutdown
+}
+
+function generate_config() {
+	# Time zone.
```

Comment:
I wanted all the logic at the top. Down in generate_config, the only defaulting logic is that if it's not set, it won't add any line in config (except for comment). There's a difference between setting having a default value and setting not being set at all.